File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.cpp
class Solution
{
public:
    string sortVowels(string s)
    {
        string ans = "";
        string l = "AaEeIiOoUu";
        string res = "";
        for (char i : s)
        {
            if (l.find(i) != std::string::npos)res.push_back(i);
        }
    sort(res.begin(), res.end());
    int i = 0, j = 0;
    while (i < s.length())
    {
        if (l.find(s[i]) != std::string::npos)
        {
            ans += res[j];
            j++;
        } 
        else ans += s[i];
        i++;
    }
    
    return ans;
    }
};

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        ans = ""
        l  = ['A','a','E','e','I','i','O','o','U','u']
        res = []
        for i in s:
            if i in l:res.append(i)
        res.sort()
        i,j = 0,0
        while(i<len(s)):
            if s[i] in l:
                ans += res[j]
                j += 1
            else:
                ans += s[i]
            i += 1
        return ans